### PR TITLE
feat: payment request

### PR DESCRIPTION
**Pull request by anhnh12**
**What does this PR do?**
- [x] New APIs
1. `POST /api/v1/payment-requests`: Generate payment code
2. `POST /api/v1/payment-requests/:code/claim`: Claim payment
3. `POST /api/v1/payment-requests`: Get payment by code
 - [x] cmd `checkexpiredpayment`

### CODE REVIEW

File: cmd/checkexpiredpayment/main.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	tx, fn := db.Store.NewTx()
+	pr.Status = constant.PayRequestExpired
+	err := tx.PayRequest.Upsert(&pr)
+	if err != nil {
+		l.Error(err, "[checkCodeExpiration] tx.PayRequest.Upsert() failed")
+		fn.Rollback(err)
+		return err
+	}
+	// credit sender balance
+	err = tx.Balance.UpsertBatches([]model.Transaction{{TokenId: pr.TokenId, ProfileId: pr.ProfileId, Amount: pr.Amount}})
```
Comment (huynguyenh) [inline cmd/checkexpiredpayment/main.go:61]: move this to controller

File: internal/app/api/handler/payrequest/pay_request.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	}
+
+	pr, err := h.controller.PayRequest.GenerateCode(req)
+	if err != nil {
+		logger.Error(err, "[handler.GenerateCode] controller.PayRequest.GenerateCode() failed")
+		c.JSON(http.StatusInternalServerError, apierror.New(err.Error(), 500, apierror.Code500))
+		return
+	}
+
+	c.JSON(http.StatusOK, view.ToRespData(pr))
```
Comment (huynguyenh) [inline internal/app/api/handler/payrequest/pay_request.go:49]: we either: 

1) change function name to ToPayRequestResponseData()
2) we create sub package `payview.ToResponseData`

File: internal/controller/payrequest/pay_request.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+	})
+
+	token, err := p.params.DB().Token.GetBySymbol(req.Token)
+	if err != nil {
+		logger.Error(err, "[GenerateCode] db.Token.GetBySymbol() failed")
+		return nil, err
+	}
+	// check balance
+	balance, err := p.params.DB().Balance.GetBalanceByTokenID(req.ProfileId, token.Id)
+	if err != nil {
```
Comment (huynguyenh) [inline internal/controller/payrequest/pay_request.go:43]: do we have any case where balance not found ?

File: internal/controller/payrequest/pay_request.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+		logger.Error(err, "[GenerateCode] db.Balance.GetBalanceByTokenID() failed")
+		return nil, err
+	}
+	amount := bigintutils.FloatToBigIntString(req.Amount, token.Decimal)
+	amt, _ := new(big.Int).SetString(amount, 10)
+	bal, _ := new(big.Int).SetString(balance.Amount, 10)
```
Comment (huynguyenh) [inline internal/controller/payrequest/pay_request.go:49]: 3 of this should into 1 other bigintutils

File: internal/controller/payrequest/pay_request.go [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+	if bal.Cmp(amt) == -1 {
```
Comment (huynguyenh) [inline internal/controller/payrequest/pay_request.go:50]: add comment

File: internal/controller/payrequest/pay_request.go [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+		err = errors.New("insufficient balance")
+		logger.Error(err, "[GenerateCode] db.Balance.GetBalanceByTokenID() failed")
+		return nil, err
+	}
+	// gen code
+	code, err := p.generateUniqueCode(10)
```
Comment (huynguyenh) [inline internal/controller/payrequest/pay_request.go:56]: do we need to check code exists?

File: internal/stringutils/string.go [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package stringutils
+
+import (
+	"math/rand"
+	"time"
+)
+
+const letters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+func GenerateCode(length int) string {
```
Comment (huynguyenh) [inline internal/stringutils/string.go:10]: add comment: `// TODO: add test`

File: internal/controller/payrequest/pay_request.go (hunk 6 above)
Comment (anhnh12) [reply internal/controller/payrequest/pay_request.go:56]: already had inside `generateUniqueCode()`

File: internal/controller/payrequest/pay_request.go (hunk 3 above)
Comment (anhnh12) [reply internal/controller/payrequest/pay_request.go:43]: yes and if will return error

File: cmd/checkexpiredpayment/main.go (hunk 1 above)
Comment (anhnh12) [reply cmd/checkexpiredpayment/main.go]: pls check `HandleExpiredPayment()`

**Pull request merged**